platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/FilterInjector.java

```diff
@@ -78,13 +94,60 @@ public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo
     }
   }
 
+  public void init() {
+    executorService.schedule(this::checkForMissedServletContexts, 1, TimeUnit.SECONDS);
```

Comment:
✏️ I see nothing that stops this service, other than the `destroy` method. That means we're going to have this thread running once a second for the lifetime of the server. That seems excessive.

Perhaps a backoff/speedup mechanism would be in order, where we adjust the time between checks. There are a few ways to do that - `UnavailableUrls` provides one example - another, probably simpler, method would be to change from using a `ScheduledExecutorService` to using a `ScheduledThreadPoolExecutor`. In that implementation, the check task would be submitted initially with a delay of 1 second and the `checkForMissedServletContexts` method would re-add the task to the pool in a finally block. If no missing services had been found on the run, the delay could be doubled (up to a maximum of say 60 seconds); if any service had been found on the run, the delay could be reset to 1 second.

This way, under a steady state system, the check would only run once every ceiling-seconds but if changes started occurring, it would check more actively.

Reply:
This particular use of the ScheduledExecutorService is a one shot execution. I put this in a separate delayed thread so as to not hold up the starting of the FilterInjector. Once the filter injector is up and running we only need to check once for ServletContexts it missed as anything new should be handled by the normal logic.